FBSimulatorControl/Model/FBProcessInfo.m

```diff
@@ -8,124 +8,89 @@
+  return self;
 }
 
 - (NSUInteger)hash
 {
-  return ((unsigned long) self.processIdentifier) | self.launchConfiguration.hash | self.launchConfiguration.hash | self.diagnostics.hash;
+  return ((unsigned long) self.processIdentifier) | self.launchPath.hash ^ self.arguments.hash | self.environment.hash;
 }
 
-- (BOOL)isEqual:(FBUserLaunchedProcess *)object
+- (BOOL)isEqual:(FBProcessInfo *)object
 {
   if (![object isKindOfClass:self.class]) {
     return NO;
   }
   return self.processIdentifier == object.processIdentifier &&
-        [self.launchConfiguration isEqual:object.launchConfiguration] &&
-        [self.launchDate isEqual:object.launchDate] &&
-        [self.diagnostics isEqual:object.diagnostics];
+        [self.launchPath isEqual:object.launchPath] &&
+        [self.arguments isEqual:object.arguments] &&
+        [self.environment isEqual:object.environment];
 }
 
 - (NSString *)description
 {
-  return [self shortDescription];
-}
-
-- (NSString *)longDescription
-{
   return [NSString stringWithFormat:
-    @"Launch %@ | PID %d | Launched %@ | Diagnostics %@",
-    self.launchConfiguration,
+    @"Process %@ | PID %d | Arguments %@ | Environment %@",
+    self.launchPath,
     self.processIdentifier,
-    self.launchDate,
-    self.diagnostics
+    self.arguments,
+    self.environment
   ];
 }
 
-- (NSString *)shortDescription
-{
-  return [NSString stringWithFormat:
-    @"Process %@ | PID %d",
-    self.launchConfiguration.shortDescription,
-    self.processIdentifier
-  ];
-}
-
-@end
-
-@implementation FBFoundProcess
-
-@synthesize launchPath = _launchPath;
-@synthesize processIdentifier = _processIdentifier;
-@synthesize arguments = _arguments;
-@synthesize environment = _environment;
+#pragma mark NSCopying
 
 - (instancetype)copyWithZone:(NSZone *)zone
 {
-  FBFoundProcess *process = [self.class new];
-  process.processIdentifier = self.processIdentifier;
-  process.launchPath = self.launchPath;
-  process.arguments = self.arguments;
-  process.environment = self.environment;
-  return process;
+  return [[self.class alloc] initWithProcessIdentifier:self.processIdentifier launchPath:self.launchPath arguments:self.arguments environment:self.environment];
 }
 
-- (NSUInteger)hash
-{
-  return ((unsigned long) self.processIdentifier) | self.launchPath.hash | self.arguments.hash | self.environment.hash;
-}
+#pragma mark NSCoding
 
-- (BOOL)isEqual:(FBUserLaunchedProcess *)object
+- (instancetype)initWithCoder:(NSCoder *)coder
 {
-  if (![object isKindOfClass:self.class]) {
-    return NO;
+  self = [super init];
+  if (!self) {
+    return nil;
   }
-  return self.processIdentifier == object.processIdentifier &&
-         [self.launchPath isEqualToString:object.launchPath] &&
-         [self.arguments isEqualToArray:object.arguments] &&
-         [self.environment isEqualToDictionary:object.environment];
+
+  _processIdentifier = [coder decodeInt32ForKey:NSStringFromSelector(@selector(processIdentifier))];
```

Comment:
I'm not too worried about migration right now, so even if the 'schema' changes and the decoding fails we can just start again.